packages/salesforcedx-utils-vscode/test/jest/helpers/paths.test.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from 'path';
+import { projectPaths, workspaceUtils } from '../../../src/';
+
+jest.mock('@salesforce/core', () => {
+  return {
+    Global: {
+      SFDX_STATE_FOLDER: '.sfdx'
+    }
+  };
+});
+
+describe('test project paths', () => {
+  const FAKE_WORKSPACE = '/here/is/a/fake/path/to/';
+
+  describe('test stateFolder', () => {
+    const FAKE_STATE_FOLDER = '.sfdx';
+    let getRootWorkspacePathStub: jest.SpyInstance;
+    let hasRootWorkspaceStub: jest.SpyInstance;
+
+    beforeEach(() => {
+      getRootWorkspacePathStub = jest.spyOn(workspaceUtils, 'getRootWorkspacePath').mockReturnValue(FAKE_WORKSPACE);
+    });
+
+    it('is defined', () => {
+      expect(projectPaths.stateFolder).toBeDefined();
+    });
+
+    it('returns path to the state folder if the project has a root workspace', () => {
+      hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace').mockReturnValue(true);
+      expect(projectPaths.stateFolder()).toEqual(path.join(FAKE_WORKSPACE, FAKE_STATE_FOLDER));
+    });
+
+    it('returns path to the state folder if the project does not have a root workspace', () => {
+      hasRootWorkspaceStub = jest.spyOn(workspaceUtils, 'hasRootWorkspace').mockReturnValue(false);
+      expect(projectPaths.stateFolder()).toEqual('');
+    });
+
+  });
+  describe('test sfdxProjectConfig', () => {
+    let stateFolderStub: jest.SpyInstance;
+    const FAKE_CONFIG = path.join(FAKE_WORKSPACE, 'sfdx-config.json');
+
+    beforeEach(() => {
+      stateFolderStub = jest.spyOn(projectPaths, 'stateFolder').mockReturnValue(FAKE_WORKSPACE);
+    });
+
+    it('is defined', () => {
+      expect(projectPaths.sfdxProjectConfig).toBeDefined();
+    });
+
+    // todo: this is failing as it appears stateFolder is not being replaced.
```

Comment:
@gbockus-sf this was being super silly, and it looked like the stateFolder() function wasn't being mocked. Any thoughts on why that simple setup might be failing?

Reply:
👀

Reply:
Tricky tricky.  So in your test you're mocking projectPaths.stateFolder, but in the code under test it's calling teh stateFolder function by name verse via the projectPaths object.  You just need to update the code under test and you'll be in business 
```function sfdxProjectConfig(): string {
  const pathToSFDXProjectConfig = path.join(
    projectPaths.stateFolder(),
    SFDX_CONFIG
  );
  return pathToSFDXProjectConfig;
}
```

Reply:
Ohhh that is tricky! Let me push what I think this should be and let me know if it looks right